src/routes/event/details.tsx

```diff
@@ -0,0 +1,96 @@
+import { useLoaderData, useNavigate } from "react-router-dom";
+import { database, auth } from "../../firebase";
+import { get, ref } from "firebase/database";
+import { Button, Container, Stack, TextField, Typography } from "@mui/material";
+import { useEffect, useState } from "react";
+
+export async function loader({ params }) {
+  return params.eventId;
+}
+
+type EventDetailsProps = {
+  eventCategory: string;
+  eventDescription: string;
+  eventEndDate: string;
+  eventLocation: string;
+  eventOrganizer: string;
+  eventStartDate: string;
+};
+
+type EventMetadataProps = {
+  eventName: string;
+};
+
+export default function EventDetails() {
+  const eventId = useLoaderData();
```

Comment:
you should be able to just use something like "useQuery" or "useParams" as far as i remember.
Never seen it done this way. 
The "loader" function is a bit misleading in its name on line 7.